static/js/publisher/metrics/graphs/activeDevices.js

```diff
@@ -3,418 +3,730 @@ import { format } from "d3-format";
+        this.height - this.padding.top - this.padding.bottom,
+        this.padding.top
+      ])
+      .nice()
+      .domain([0, this.maxYValue + Math.ceil(this.maxYValue * 0.1)]);
   }
 
-  // If we've added anything to the 'other' series,
-  // add it to the array
-  if (other.value > 0) {
-    dataArr.push(other);
+  prepareSVG() {
+    // this.svg.attr("width", this.holder.clientWidth);
+    this.svg.selectAll("*").remove();
+
+    this.width =
+      this.holder.clientWidth -
+      this.margin.left -
+      this.margin.right -
+      this.padding.left -
+      this.padding.right;
+
+    this.height =
+      this.svg.attr("height") -
+      this.margin.top -
+      this.margin.bottom -
+      this.padding.top -
+      this.padding.bottom;
+
+    this.g = this.svg
+      .append("g")
+      .attr("transform", `translate(${this.margin.left},${this.margin.top})`);
+
+    return this;
   }
 
-  return dataArr
-    .map(item => {
-      if (!item.skip) {
-        return [
-          `<span class="snapcraft-graph-tooltip__series${
-            item.key === currentHoverKey ? " is-hovered" : ""
-          }" title="${item.key}">`,
-          `<span class="snapcraft-graph-tooltip__series-name">${
-            item.key
-          }</span>`,
-          `<span class="snapcraft-graph-tooltip__series-color"${
-            !item.count ? `style="background:${colorScale(item.key)};"` : ""
-          }></span>`,
-          `<span class="snapcraft-graph-tooltip__series-value">${commaValue(
-            item.value
-          )}</span>`,
-          `</span>`
-        ].join("");
-      }
-    })
-    .join("");
-};
-
-const tooltipTemplate = (data, currentHoverKey, colorScale) => {
-  return [
-    `<div class="p-tooltip p-tooltip--top-center" style="display: block;">`,
-    `<span class="p-tooltip__message" role="tooltip" style="display: block;">`,
-    `<span class="snapcraft-graph-tooltip__title">${tooltipTimeFormat(
-      data.date
-    )}</span>`,
-    tooltipRows(data, currentHoverKey, colorScale),
-    `</span>`,
-    `</div>`
-  ].join("");
-};
-
-function drawGraph(holderSelector, holder, activeDevices, annotations) {
-  // Basic svg setup
-  const svg = select(`${holderSelector} svg`);
-  svg.attr("width", holder.clientWidth);
-  svg.selectAll("*").remove();
-
-  const margin = { top: 20, right: 0, bottom: 30, left: 50 };
-  const padding = { top: 0, right: 0, bottom: 16, left: 16 };
-  const width =
-    svg.attr("width") -
-    margin.left -
-    margin.right -
-    padding.left -
-    padding.right;
-  const height =
-    svg.attr("height") -
-    margin.top -
-    margin.bottom -
-    padding.top -
-    padding.bottom;
-  const g = svg
-    .append("g")
-    .attr("transform", `translate(${margin.left},${margin.top})`);
-
-  // Prepare the data
-  const _data = [];
-  const _keys = [];
-  let max = 0;
-  activeDevices.buckets.forEach((bucket, i) => {
-    let obj = {
-      date: utcParse("%Y-%m-%d")(bucket)
+  prepareStackedData() {
```

Comment:
Do all those methods need to be public? It's hard to discover what is meant to be a public API for `Graph` and what is just internal methods.